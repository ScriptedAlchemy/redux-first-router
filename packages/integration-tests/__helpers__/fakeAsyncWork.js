export default (cb) =>
  new Promise((resolve) => {
    setTimeout(() => resolve(cb), 1)
  })
