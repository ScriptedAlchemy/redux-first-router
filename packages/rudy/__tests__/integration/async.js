import createTest from '../../__helpers__/createTest'
import fakeAsyncWork from '../../__helpers__/fakeAsyncWork'

createTest('callbacks sequentially run as promises', {
  SECOND: {
    path: '/second',
    beforeEnter: async () => {
      await fakeAsyncWork()
    },
    thunk: async () => {
      await fakeAsyncWork()
    },
    onComplete: async () => {
      await fakeAsyncWork()
      return 'payload'
    },
  },
})

createTest('callbacks sequentially run as promises /w redirect', {
  SECOND: {
    path: '/second',
    beforeEnter: async () => {
      await fakeAsyncWork()
    },
    thunk: async ({ action }) => {
      await fakeAsyncWork()
      if (action.type !== 'SECOND') return
      return { type: 'REDIRECTED' }
    },
    onComplete: async () => {
      await fakeAsyncWork()
    },
  },
})
