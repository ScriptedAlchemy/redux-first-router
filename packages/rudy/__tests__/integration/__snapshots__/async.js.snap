// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`callbacks sequentially run as promises /w redirect: 1 - response 1`] = `
Object {
  "payload": "redirect_complete",
  "type": "REDIRECTED_COMPLETE",
}
`;

exports[`callbacks sequentially run as promises /w redirect: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`callbacks sequentially run as promises /w redirect: 1 - routes - REDIRECTED - onComplete 1`] = `1`;

exports[`callbacks sequentially run as promises /w redirect: 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`callbacks sequentially run as promises /w redirect: 1 - routes - SECOND - onComplete 1`] = `0`;

exports[`callbacks sequentially run as promises /w redirect: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`callbacks sequentially run as promises /w redirect: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/redirected",
          "scene": "",
          "search": "",
          "url": "/redirected",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "REDIRECTED",
      },
    ],
    "from": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 1,
        "key": "345678",
        "pathname": "/second",
        "scene": "",
        "search": "",
        "url": "/second",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "SECOND",
    },
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "replace",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/redirected",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 302,
    "type": "REDIRECTED",
    "universal": false,
    "url": "/redirected",
  },
  "title": "REDIRECTED_COMPLETE - \\"redirect_complete\\"",
}
`;

exports[`callbacks sequentially run as promises /w redirect: 1 - title 1`] = `"REDIRECTED"`;

exports[`callbacks sequentially run as promises: 1 - response 1`] = `
Object {
  "payload": "payload",
  "type": "SECOND_COMPLETE",
}
`;

exports[`callbacks sequentially run as promises: 1 - routes - NEVER_USED_PATHLESS - thunk 1`] = `0`;

exports[`callbacks sequentially run as promises: 1 - routes - REDIRECTED - onComplete 1`] = `0`;

exports[`callbacks sequentially run as promises: 1 - routes - SECOND - beforeEnter 1`] = `1`;

exports[`callbacks sequentially run as promises: 1 - routes - SECOND - onComplete 1`] = `1`;

exports[`callbacks sequentially run as promises: 1 - routes - SECOND - thunk 1`] = `1`;

exports[`callbacks sequentially run as promises: 1 - state 1`] = `
Object {
  "location": Object {
    "basename": "",
    "blocked": null,
    "direction": "forward",
    "entries": Array [
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/first",
          "scene": "",
          "search": "",
          "url": "/first",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "FIRST",
      },
      Object {
        "basename": "",
        "hash": "",
        "location": Object {
          "key": "345678",
          "pathname": "/second",
          "scene": "",
          "search": "",
          "url": "/second",
        },
        "params": Object {},
        "query": Object {},
        "state": Object {},
        "type": "SECOND",
      },
    ],
    "from": null,
    "hash": "",
    "index": 1,
    "key": "345678",
    "kind": "push",
    "length": 2,
    "n": 1,
    "params": Object {},
    "pathname": "/second",
    "pop": false,
    "prev": Object {
      "basename": "",
      "hash": "",
      "location": Object {
        "index": 0,
        "key": "345678",
        "pathname": "/first",
        "scene": "",
        "search": "",
        "url": "/first",
      },
      "params": Object {},
      "query": Object {},
      "state": Object {},
      "type": "FIRST",
    },
    "query": Object {},
    "ready": true,
    "scene": "",
    "search": "",
    "state": Object {},
    "status": 200,
    "type": "SECOND",
    "universal": false,
    "url": "/second",
  },
  "title": "SECOND_COMPLETE - \\"payload\\"",
}
`;

exports[`callbacks sequentially run as promises: 1 - title 1`] = `"SECOND"`;
